### PR TITLE
Breaking changes - migrations, multi-commands and more experimental changes

### PR DESCRIPTION
**Pull request by kabirbaidhya**
- [x] Migrations support (SQL based, inspired by [go-migrate](https://github.com/golang-migrate/migrate) but fully based on [Knex's API](http://knexjs.org/#Migrations-API))
- [x] Multi-commands CLI
- [x] Other significant updates / several breaking changes
- [x] Custom connection resolver support (#69)

---

**Screenshots**
image
image
image

### CODE REVIEW

**Comment by kabirbaidhya** [comment]
@mesaugat @SafalPandey @cham11ng Heads up - tons of changes. 
You may skim through the code and leave comments but this is still work-in-progress / experimental; I'll share a tutorial for this once finished and we can go for final review and beta test.

**Comment by mesaugat** [comment]
@kabirbaidhya Are we not using this version already? Should we merge this? And what about the other PR #42?

**Comment by kabirbaidhya** [comment]
@mesaugat Yes, we are. The beta/experimental versions are released but we have been using it in production already since few months and its kind of stable. 

You can ignore #42 change right now - it's completely independent and could be worked on later after this merge.

File: src/domain/Configuration.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    post_sync: string[];
+  };
+  injectedConfig: {
+    vars: Mapping<string>;
+  };
+  migration: {
+    directory: string;
+    tableName: string;
+    // TODO: Only 'sql' is supported sourceType now. JS will be supported later.
+    sourceType: 'sql' | 'javascript';
```
Comment (kabirbaidhya) [inline src/domain/Configuration.ts:22]: @mesaugat This `javascript` based migration file support is currently missing. We can merge this and add that support soon in the next release. 

All we need to add here, is an ability to import { up, down } functions from the migration files. It won't be that hard, given all the groundwork is already done here.

Comment (mesaugat) [reply src/domain/Configuration.ts:22]: Nice.

**Review by mesaugat — APPROVED** [review]
Merging, didn't look at all the details. We have been already using this in production.

**Pull request merged**